sympy/functions/elementary/hyperbolic.py

```diff
@@ -1378,17 +1390,24 @@ def taylor_term(n, x, *previous_terms):
                 F = factorial(k)
                 return -R / F * S.ImaginaryUnit * x**n / n
 
-    def _eval_as_leading_term(self, x, logx=None, cdir=0):
-        from sympy.series.order import Order
-        arg = self.args[0].as_leading_term(x)
-
-        if x in arg.free_symbols and Order(1, x).contains(arg):
-            return S.ImaginaryUnit*S.Pi/2
-        else:
-            return self.func(arg)
+    def _eval_as_leading_term(self, x, logx=None, cdir=0):  # acosh
+        arg = self.args[0]
+        x0 = arg.subs(x, 0).cancel()
+        # Handling Branch cuts (-oo, 1)
+        if x0 in (-S.One, S.Zero, S.One, S.ComplexInfinity):
+            return self.rewrite(log)._eval_as_leading_term(x, logx=logx, cdir=cdir)
+        ndir = arg.dir(x, cdir if cdir else 1)
+        if (x0 - 1).is_negative:
+            if im(ndir).is_negative:
+                if x0.is_negative:
+                    return self.func(x0) - 2*I*pi
```

Comment:
What if `x0` is positive (and < 1)?

Reply:
Nice catch , my implementation essentially mimics a function having branch cuts from `(-oo, 0)` but for `acosh` we need  `(-oo, 1)` , so we need handling in the block you've pointed out . Will add code and tests for the same .

Reply:
Thanks for pointing this out @jksuom . I wasn't essentially taking care of of few corner cases which the latest commit fixes . So 
1) `acosh` having branch cuts from `(-oo, 1)` was being treated as `(-oo, 0)`
2) `asech` having branch cuts from `(-oo, 0] U (1, oo)` was being treated as `(-oo, -1] U (1, oo)` 
Hence I was missing some bit of portion in both . I have fixed this now by adding some handling along with test cases for both these functions . I now feel that we can answer all possible cases involved with `acosh` and `asech` correctly !